lib/options.ts

```diff
@@ -71,3 +77,35 @@ export const OPTION_DEFAULTS = {
   [OPTION_TYPE.URL_CONFIGS]: undefined,
   [OPTION_TYPE.URL_RULE_DOC]: undefined,
 } satisfies Record<OPTION_TYPE, unknown>; // Satisfies is used to ensure all options are included, but without losing type information.
+
+/** Combines provided options with default options. */
+export function getResolvedOptions(
+  options: GenerateOptions = {},
+): ResolvedGenerateOptions {
+  const check = options.check ?? OPTION_DEFAULTS[OPTION_TYPE.CHECK];
+  const configEmoji =
+    options.configEmoji ?? OPTION_DEFAULTS[OPTION_TYPE.CONFIG_EMOJI];
+  const configFormat =
+    options.configFormat ?? OPTION_DEFAULTS[OPTION_TYPE.CONFIG_FORMAT];
+  const ignoreConfig = stringOrArrayWithFallback(
+    options?.ignoreConfig,
+    OPTION_DEFAULTS[OPTION_TYPE.IGNORE_CONFIG],
+  );
+
+  // @ts-expect-error This will be filled in later with all the remaining options. This being
```

Comment:
This will be removed in my next PR. 
It turns out that `Required<GeneratedOptions>` was a bad way to do it.
I discovered a much better way to do this using type inference.